samples/previous-viewpoint/src/services/customService/CustomService.ts

```diff
@@ -0,0 +1,47 @@
+import { ServiceBase } from "@vertigis/web/services";
+import { command, MapsLike, ViewpointLike } from "@vertigis/web/messaging";
+
+export default class CustomService extends ServiceBase {
+    private _history: ViewpointLike[] = [];
+    private movedBecauseCommand = false;
+
+    @command("custom-service.push-viewpoint")
+    protected _handlePushViewpoint(args: {
+        maps: MapsLike;
+        viewpoint: ViewpointLike;
+    }): void {
+        // Don't update the history if the reason the viewpoint changed was our
+        // `goToPreviousViewPoint` command
+        if (!this.movedBecauseCommand) {
+            this._history.push(args.viewpoint);
+        }
+        this.movedBecauseCommand = false;
+    }
+
+    @command("custom-service.go-to-previous-viewpoint")
+    protected async _handleGoToPreviousViewpoint(): Promise<void> {
+        const currentViewpoint: {
+            maps: MapsLike;
+            viewpoint: ViewpointLike;
+        } = (await this.messages.operations.map.getViewpoint.execute()) as any as {
+            maps: MapsLike;
+            viewpoint: ViewpointLike;
+        };
+        let previousViewpoint = this._history.pop();
+        while (
```

Comment:
Under what circumstances would the current viewpoint be the identical viewpoint to something pushed onto the history stack, when the push command is wired to `onViewpointChanged`? It would be worth a comment here explaining why it's needed, or else just get rid of the loop.

Reply:
I added a comment and removed the loop.